geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -1781,4 +1784,43 @@ ParallelGatewaySenderQueueMetaRegion newMetataRegion(InternalCache cache, final
       return meta;
     }
   }
+
+  public void clearQueue() {
+
+    this.sender.pause();// it wil take internal read-write-lock
+    try {
+      for (PartitionedRegion prQ : this.userRegionNameToshadowPRMap.values()) {
+        clearPartitionedRegion((PartitionedRegion) prQ);
+      }
+    } finally {
+      if (this.sender.isPaused())
```

Comment:
Shouldn't this be true always since we paused the sender in line 1790?  If not, what happens if another thread does unpause the sender, how will all of this clear logic behave?

Reply:
ConcurrentParallelGatewaySenderQueue.java i have added lock so no other will held that.

Reply:
So you are saying that no one else can unpause it due to holding the lock.  So we can remove the check to see if sender is paused?